_rules/widget-non-empty-accessible-name-rdzs6q.md

```diff
@@ -0,0 +1,382 @@
+---
+id: rdzs6q
+name: Widget has non-empty accessible name
+rule_type: atomic
+description: |
+  This rule checks that each widget has a non-empty accessible name.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Carlos Duarte
+  previous_authors:
+    - Anne Thyme Nørregaard
+    - Bryn Anderson
+    - Stein Erik Skotkjerra
+    - Wilco Fiers
+  assets:
+    - Image used in passed example 14 is courtesy of NASA/JPL-Caltech.
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+---
+
+## Applicability
+
+This rule applies to any element that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has a [semantic role][] that is or inherits from one of the following [semantic roles][]:
+
+- `button`; or
```

Comment:
I think we're going to have problems with image buttons if we do this. If we added an image button example to this rule, an implementation that would report that as failing 1.1.1 would not be considered a consistent implementation of this rule, per the new definitions we're working on for consistency (see https://github.com/w3c/wcag-act/pull/522).

I'm not sure how to handle this case yet. But I would suggest we discuss this on an upcoming call. I'd like us to carefully work through the implications of not having an image button exception here.